cpp/src/io/parquet/reader_impl_preprocess.cu

```diff
@@ -733,39 +727,54 @@ void reader::impl::load_and_decompress_data(std::vector<row_group_info> const& r
         total_decompressed_size += col_meta.total_uncompressed_size;
       }
     }
-    // Read compressed chunk data to device memory
-    read_rowgroup_tasks.push_back(read_column_chunks_async(_sources,
-                                                           raw_page_data,
-                                                           chunks,
-                                                           io_chunk_idx,
-                                                           chunks.size(),
-                                                           column_chunk_offsets,
-                                                           chunk_source_map,
-                                                           _stream));
-
     remaining_rows -= row_group.num_rows;
   }
+
+  // Read compressed chunk data to device memory
+  read_rowgroup_tasks.push_back(read_column_chunks_async(_sources,
+                                                         raw_page_data,
+                                                         chunks,
+                                                         0,
+                                                         chunks.size(),
+                                                         column_chunk_offsets,
+                                                         chunk_source_map,
+                                                         _stream));
+
+  CUDF_EXPECTS(remaining_rows <= 0, "All rows data must be read.");
```

Comment:
Should this be this?
```suggestion
  CUDF_EXPECTS(remaining_rows == 0, "All rows data must be read.");
```

Because if `remaining_rows < 0` we probably "over-read" (i.e., wrongly read more rows than the file has)?

Reply:
Turns out this isn't going to work as-is. I agree that it should be a zero check here, but we're subtracting `row_group.num_rows` each iteration, but we may not have read that many rows.

```
auto const row_group_rows   = std::min<int>(remaining_rows, row_group.num_rows);
```

Still thinking about the right path here.